### DISCUSSION TITLE
v2.6.1 Released

### DISCUSSION THREAD

**Discussion by iBotPeaches** [iBotPeaches/Apktool#2766] [Releases]
## What's Changed
* Begin options refactor by @iBotPeaches in https://github.com/iBotPeaches/Apktool/pull/2648
* fix: decoding references to private resources by @MrIkso in https://github.com/iBotPeaches/Apktool/pull/2650
* fix: support for signature scheme v4 by @iBotPeaches in https://github.com/iBotPeaches/Apktool/pull/2705
* Replace use of deprecated methods with their recommended replacements by @alsutton in https://github.com/iBotPeaches/Apktool/pull/2713
* Update dependencies by @Goooler in https://github.com/iBotPeaches/Apktool/pull/2715
* Cleanup trailing temp files/folders by @iBotPeaches in https://github.com/iBotPeaches/Apktool/pull/2742
* fix: prevent file hold on tmpDir during build by @iBotPeaches in https://github.com/iBotPeaches/Apktool/pull/2745
* refactor: drop unused exceptions by @iBotPeaches in https://github.com/iBotPeaches/Apktool/pull/2746
* Preventing instantiation of untrusted classes. by @iBotPeaches in https://github.com/iBotPeaches/Apktool/pull/2760

## New Contributors
* @MrIkso made their first contribution in https://github.com/iBotPeaches/Apktool/pull/2650
* @alsutton made their first contribution in https://github.com/iBotPeaches/Apktool/pull/2713

**Full Changelog**: https://github.com/iBotPeaches/Apktool/compare/v2.6.0...v2.6.1
**Release Post**: https://connortumbleson.com/2022/02/26/apktool-v2-6-1-released/

<hr /><em>This discussion was created from the release <a href='https://github.com/iBotPeaches/Apktool/releases/tag/v2.6.1'>v2.6.1 Released</a>.</em>

**Comment by Prince01github** [comment]
Sir we're facing a lot of problems while Rebuilding resourcesUsing APK template: /sdcard/fb.apk
[-] No platform was selected, choosing Msf::Module::Platform::Android from the payload
[-] No arch selected, selecting arch: dalvik from the payload
[*] Creating signing key and keystore..
[*] Decompiling original APK..
[*] Decompiling payload APK..
[*] Locating hook point..
[*] Adding payload as package com.facebook.lite.gjvak
[*] Loading /tmp/d20220314-29626-1xsp3wf/original/smali/com/facebook/lite/ClientApplicationSplittedShell.smali and injecting payload..
[*] Poisoning the manifest with meterpreter permissions..
[*] Adding <uses-permission android:name="android.permission.WRITE_SETTINGS"/>
[*] Adding <uses-permission android:name="android.permission.READ_CALL_LOG"/>
[*] Adding <uses-permission android:name="android.permission.SEND_SMS"/>
[*] Adding <uses-permission android:name="android.permission.READ_SMS"/>
[*] Adding <uses-permission android:name="android.permission.WRITE_CALL_LOG"/>
[*] Adding <uses-permission android:name="android.permission.RECEIVE_SMS"/>
[*] Adding <uses-permission android:name="android.permission.SET_WALLPAPER"/>
[*] Rebuilding apk with meterpreter injection as /tmp/d20220314-29626-1xsp3wf/output.apk
[-] I: Using Apktool 2.6.1
I: Checking whether sources has changed...
I: Smaling smali folder into classes.dex...
I: Checking whether resources has changed...
I: Building resources...
brut.androlib.AndrolibException: brut.common.BrutException: could not exec (exit code = 132): [/tmp/brut_util_Jar_69221135822979910956479523302603090470.tmp, p, --forced-package-id, 127, --min-sdk-version, 15, --target-sdk-version, 31, --version-code, 354975467, --version-name, 293.0.0.9.114, --no-version-vectors, -F, /tmp/APKTOOL5540271027132335925.tmp, -e, /tmp/APKTOOL3082563400059542143.tmp, -0, arsc, -I, /home/kali/.local/share/apktool/framework/1.apk, -S, /tmp/d20220314-29626-1xsp3wf/original/res, -M, /tmp/d20220314-29626-1xsp3wf/original/AndroidManifest.xml]
Error: Unable to rebuild apk with apktool

I'LL BE GLAD IF YOU CAN FIX THIS ERROR

**Comment by seisdr** [comment]
```
 apktool  b app-debug/ k.apk                                                     I: Using Apktool 2.6.1-dirty                                                            I: Checking whether sources has changed...                                              I: Checking whether resources has changed...                                            I: Building resources...                                                                W: res/drawable-v21/$avd_hide_password__0.xml: Invalid file name: must contain only [a-z0-9_.]                                                                                  W: res/drawable-v21/$avd_hide_password__1.xml: Invalid file name: must contain only [a-z0-9_.]                                                                                  W: res/drawable-v21/$avd_hide_password__2.xml: Invalid file name: must contain only [a-z0-9_.]                                                                                  W: res/drawable-v21/$avd_show_password__0.xml: Invalid file name: must contain only [a-z0-9_.]                                                                                  W: res/drawable-v21/$avd_show_password__1.xml: Invalid file name: must contain only [a-z0-9_.]
W: res/drawable-v21/$avd_show_password__2.xml: Invalid file name: must contain only [a-z0-9_.]                                                                                  brut.androlib.AndrolibException: brut.common.BrutException: could not exec (exit code = 1): [/data/data/com.termux/files/usr/tmp/brut_util_Jar_25626272608559812262530433451988827701.tmp, p, --min-sdk-version, 19, --target-sdk-version, 27, --version-code, 1, --version-name, 0.0.2, --no-version-vectors, -F, /data/data/com.termux/files/usr/tmp/APKTOOL15287739289616313664.tmp, -0, resources.arsc, -0, META-INF/android.arch.core_runtime.version, -0, META-INF/android.arch.lifecycle_livedata-core.version, -0, META-INF/android.arch.lifecycle_runtime.version, -0, META-INF/android.arch.lifecycle_viewmodel.version, -0, META-INF/com.android.support_animated-vector-drawable.version, -0, META-INF/com.android.support_appcompat-v7.version, -0, META-INF/com.android.support_design.version, -0, META-INF/com.android.support_recyclerview-v7.version, -0, META-INF/com.android.support_support-compat.version, -0, META-INF/com.android.support_support-core-ui.version, -0, META-INF/com.android.support_support-core-utils.version, -0, META-INF/com.android.support_support-fragment.version, -0, META-INF/com.android.support_support-media-compat.version, -0, META-INF/com.android.support_support-v4.version, -0, META-INF/com.android.support_support-vector-drawable.version, -0, META-INF/com.android.support_transition.version, -0, wav, -0, assets/tmp/l.txt, -0, png, -0, arsc, -I, /data/data/com.termux/files/home/.local/share/apktool/framework/1.apk, -S, /data/data/com.termux/files/home/apk/app-debug/res, -M, /data/data/com.termux/files/home/apk/app-debug/AndroidManifest.xml]
```
Why This is an ongoing issue

**Reply by Eddyna** [reply]
```
@Benjamin-Dobell 
```
